assets/prototype/domain/eventEditor/tickets/ticketPriceCalculator/formDecorators/usePriceTypeDecorator.js

```diff
@@ -0,0 +1,67 @@
+import { allPass, map, propEq, when } from 'ramda';
+import { isEqual, getFromFormData } from './utilities';
+import useTicketPriceCalculator from '../hooks/useTicketPriceCalculator';
+import usePriceTypes from '../../../data/queries/priceTypes/usePriceTypes';
+import { getPriceType } from '../../../../shared/predicates/prices/selectionPredicates';
+import { getDefaultPriceType } from '../../../../shared/predicates/priceTypes/selectionPredicates';
+
+const usePriceTypeDecorator = () => {
+	const priceTypes = usePriceTypes();
+	const getPriceTypeForPrice = getPriceType(priceTypes);
+	const defaultPriceType = getDefaultPriceType(priceTypes);
+	const calculator = useTicketPriceCalculator();
+	return {
+		field: /^prices\[\d+\]\.priceType$/,
+		isEqual: isEqual,
+		updates: (value, name, formData) => {
+			const pricePath = name.replace('.priceType', '');
+			const price = getFromFormData(pricePath, formData);
+			const reverseCalc = getFromFormData('ticket.reverseCalculate', formData);
+			let priceType = getPriceTypeForPrice(price);
+			priceType = priceType || defaultPriceType;
```

Comment:
I have mixed feelings about this mutation.
Maybe just:
`const priceType = getPriceTypeForPrice(price) || defaultPriceType;` ?

Reply:
oh ya, that's better... will do that